datafusion/execution/src/cache/list_files_cache.rs

```diff
@@ -172,20 +172,73 @@ impl DefaultListFilesCacheState {
         }
     }
 
-    /// Returns the respective entry from the cache, if it exists and the entry
-    /// has not expired by `now`.
+    /// Performs a prefix-aware cache lookup.
     ///
-    /// If the entry exists it becomes the most recently used. If the entry has expired it is
-    /// removed from the cache
-    fn get(&mut self, key: &Path, now: Instant) -> Option<Arc<Vec<ObjectMeta>>> {
-        let entry = self.lru_queue.get(key)?;
+    /// # Arguments
+    /// * `table_base` - The table's base path (the cache key)
+    /// * `prefix` - Optional prefix filter relative to the table base path
+    /// * `now` - Current time for expiration checking
+    ///
+    /// # Behavior
+    /// - Fetches the cache entry for `table_base`
+    /// - If `prefix` is `Some`, filters results to only files matching `table_base/prefix`
+    /// - Returns the (potentially filtered) results
+    ///
+    /// # Example
+    /// ```text
+    /// get_with_prefix("my_table", Some("a=1"), now)
+    ///   → Fetch cache entry for "my_table"
+    ///   → Filter to files matching "my_table/a=1/*"
+    ///   → Return filtered results
+    /// ```
+    fn get_with_prefix(
+        &mut self,
+        table_base: &Path,
+        prefix: Option<&Path>,
+        now: Instant,
+    ) -> Option<Arc<Vec<ObjectMeta>>> {
+        let entry = self.lru_queue.get(table_base)?;
 
-        match entry.expires {
-            Some(exp) if now > exp => {
-                self.remove(key);
-                None
+        // Check expiration
+        if let Some(exp) = entry.expires
+            && now > exp
+        {
+            self.remove(table_base);
+            return None;
+        }
+
+        match prefix {
+            None => {
+                // No prefix filter, return all files
+                Some(Arc::clone(&entry.metas))
+            }
+            Some(prefix) => {
+                // Build the full prefix path: table_base/prefix
+                let full_prefix = if table_base.as_ref().is_empty() {
```

Comment:
Style nit -- I think this code would format a bit better using a `let-else` statement rather than a `match`:

```rust
let Some(prefix) = prefix else {
    return Some(Arc::clone(&entry.metas))
}

let full_prefix = ...
```